### PR TITLE
A couple loop optimizations

### PR DESCRIPTION
**Pull request by txanatan**
These functions were iterating over things twice when they didn't need to, resulting in a speed decrease (however minimal). This cleans those up and makes the code easier to follow.

### CODE REVIEW

File: sappho/tilemap.py [hunk 1]
```diff
@@ -67,28 +67,26 @@ def parse_rules(path_to_rules_file):
...
-        with open(path_to_rules_file) as f:
-            rules = f.readlines()
-
         tile_rules = {}
 
-        for rule in rules:
-            tile_ids_affected, flags = rule.split('=')
-            tile_ids_affected = [id_ for id_ in tile_ids_affected.split(',')]
-            flags = [flag.strip() for flag in flags.split(',')]
+        with open(path_to_rules_file) as f:
```
Comment (lillian-lemmer) [inline sappho/tilemap.py]: This really shouldn't be nested...

**Comment by lillian-lemmer** [comment]
We're gonna need a meeting on this one.

File: sappho/tilemap.py (hunk 1 above)
Comment (txanatan) [inline sappho/tilemap.py]: It's faster nesting, and it makes more sense from my point of view.

Considering `file.readlines()` is equivalent to the following:

```python
for line in f:
    yield line
```

It's much faster to optimize out that full iteration when we're iterating over lines in the first place.

Comment (lillian-lemmer) [inline sappho/tilemap.py]: I can't seem to find where it says that readlines is an iterator object? It seems to just to generate a list. Everything I've read says "automatically parses the read contents into a list of lines."

I feel like that's a needless optimization and it's premature.

I wanna focus on legibility over speediness. I avoid indentation when possible.

Optimize once there's a problem with bottleneck.

**Comment by lillian-lemmer** [comment]
Imma go ahead and edit this branch.

**Pull request merged**